modules/ml/src/test/java/org/apache/ignite/ml/preprocessing/encoding/TargetEncoderPreprocessorTest.java

```diff
@@ -0,0 +1,90 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.ml.preprocessing.encoding;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.HashSet;
+import org.apache.ignite.ml.dataset.feature.extractor.Vectorizer;
+import org.apache.ignite.ml.dataset.feature.extractor.impl.DummyVectorizer;
+import org.apache.ignite.ml.math.primitives.vector.Vector;
+import org.apache.ignite.ml.math.primitives.vector.impl.DenseVector;
+import org.apache.ignite.ml.preprocessing.encoding.target.TargetEncoderPreprocessor;
+import org.apache.ignite.ml.preprocessing.encoding.target.TargetEncodingMeta;
+import org.junit.Test;
+import static org.junit.Assert.assertArrayEquals;
+
+/**
+ * Tests for {@link TargetEncoderPreprocessor}.
+ */
+public class TargetEncoderPreprocessorTest {
+    /** Tests {@code apply()} method. */
+    @Test
+    public void testApply() {
+        Vector[] data = new Vector[] {
+            new DenseVector(new Serializable[] {"1", "Moscow", "A"}),
+            new DenseVector(new Serializable[] {"2", "Moscow", "B"}),
+            new DenseVector(new Serializable[] {"3", "Moscow", "B"}),
+        };
+
+        Vectorizer<Integer, Vector, Integer, Double> vectorizer = new DummyVectorizer<>(0, 1, 2);
+
+        TargetEncoderPreprocessor<Integer, Vector> preprocessor = new TargetEncoderPreprocessor<Integer, Vector>(
+            new TargetEncodingMeta[]{
+                new TargetEncodingMeta(
+                    0.5,
+                    new HashMap() {
+                        {
+                            put("1", 1.0);
+                            put("2", 0.0);
+                        }
+                    }
+                ),
+                new TargetEncodingMeta(
+                    0.1,
+                    new HashMap() {}
+                ),
+                new TargetEncodingMeta(
+                    0.1,
+                    new HashMap() {
+                        {
+                            put("A", 1.0);
+                            put("B", 2.0);
+                        }
+                    }
+                ),
+            },
+            vectorizer,
+            new HashSet() {
+                {
+                    add(0);
+                    add(1);
+                    add(2);
+                }
+            });
+
+        double[][] postProcessedData = new double[][] {
+            {1.0, 0.1, 1.0},
```

Comment:
Could you explain please numbers in the last columns: why are they 1.0 and 2.0? not 0.33 and 0.66

Reply:
Described each case